lib/resourcemerge/batch.go

```diff
@@ -8,21 +8,26 @@ import (
 // EnsureJob ensures that the existing matches the required.
 // modified is set to true when existing had to be updated with required.
 func EnsureJob(modified *bool, existing *batchv1.Job, required batchv1.Job) {
-	EnsureObjectMeta(modified, &existing.ObjectMeta, required.ObjectMeta)
 
-	if existing.Spec.Selector == nil {
-		*modified = true
-		existing.Spec.Selector = required.Spec.Selector
-	}
-	if !equality.Semantic.DeepEqual(existing.Spec.Selector, required.Spec.Selector) {
-		*modified = true
-		existing.Spec.Selector = required.Spec.Selector
+	// To set Selector you must set ManualSelector to True. If 'required'
+	// sets Selector to value not equal to existing, let current Job
+	// finish as Selector is immutable.
+	if required.Spec.ManualSelector != nil {
+
+		// If ManualSelector is not True we don't care about Selector
+		if *required.Spec.ManualSelector {
+			if !equality.Semantic.DeepEqual(existing.Spec.Selector, required.Spec.Selector) {
+				return
+			}
+		}
+		setBoolPtr(modified, &existing.Spec.ManualSelector, required.Spec.ManualSelector)
```

Comment:
`setBoolPrt` [is a noop when the required pointer is `null`][1], so we can leave this `setBoolPtr` call alone in its previous location without changing the logic.  Can you reroll to do that?

[1]: https://github.com/openshift/cluster-version-operator/blob/abfb48da582c81937bb6a08abaef163199c68f6e/lib/resourcemerge/core.go#L511-L513